packages/task/src/browser/provided-task-configurations.ts

```diff
@@ -66,6 +66,16 @@ export class ProvidedTaskConfigurations {
         }
     }
 
+    async getTaskById(id: string): Promise<TaskConfiguration | undefined> {
```

Comment:
Looking at the code, I don't think this is going to work: when we cache tasks, two contributed tasks that have the same scope and the same label will overwrite one another: the id does not factor into the key at all.